File: backend/app/modules/news/api/news.py
from typing import Any, Dict, List

from fastapi import APIRouter, Depends

from app.api.dependencies import get_news_service
from app.modules.news.services.news import NewsService

router = APIRouter(prefix="/news", tags=["news"])

@router.get("/health")
def news_health():
    return {"module": "news", "status": "ok"}

@router.get("")
def get_all_news(
    service: NewsService = Depends(get_news_service)
) -> Dict[str, Any]:
    return service.get_all_recent(limit=30)

@router.get("/{symbol}")
def get_news_for_symbol(
    symbol: str,
    service: NewsService = Depends(get_news_service)
) -> List[Dict[str, Any]]:
    return service.get_recent_news(symbol, limit=10)

File: backend-node/src/routes/news/news.ts
import { Router } from "express";
import { getAllRecent, getRecentNews } from "../../lib/news/news";

// Port of app/modules/news/api/news.py — mounted at /api/v1/news (matches
// Python's router.prefix="/news" + app-level "/api/v1").
export const newsRouter = Router();

newsRouter.get("/health", (_req, res) => {
  res.json({ module: "news", status: "ok" });
});

newsRouter.get("/", async (_req, res, next) => {
  try {
    res.json(await getAllRecent(30));
  } catch (e) {
    next(e);
  }
});

newsRouter.get("/:symbol", async (req, res, next) => {
  try {
    res.json(await getRecentNews(req.params.symbol, 10));
  } catch (e) {
    next(e);
  }
});
